contrib/playlib/src/mill/playlib/RouterGeneratorWorker.scala

```diff
@@ -0,0 +1,117 @@
+        val compileMethod = routerCompilerClass.getMethod("compile",
+          routesCompilerTaskClass,
+          cl.loadClass("play.routes.compiler.RoutesGenerator"),
+          classOf[java.io.File])
+        val instance = new RouterGeneratorWorkerApi {
+          override def compile(task: RoutesCompilerTask, generatorType: String = "injected", generatedDir: File): Either[Seq[CompilationResult], Seq[File]] = {
+            // Since the classloader is isolated, we do not share any classes with the Mill classloader.
+            // Thus both classloaders have different copies of "scala.collection.Seq" which are not compatible.
+            val additionalImports = cl.loadClass("scala.collection.mutable.WrappedArray$ofRef")
+              .getConstructors()(0)
+              .newInstance(task.additionalImports.toArray)
+              .asInstanceOf[AnyRef]
+            val args = Array[AnyRef](task.file,
+              additionalImports,
+              Boolean.box(task.forwardsRouter),
+              Boolean.box(task.reverseRouter),
+              Boolean.box(task.namespaceReverseRouter))
+            val routesCompilerTaskInstance = routerCompilerTaskConstructor.newInstance(args: _*).asInstanceOf[Object]
+            val routesGeneratorInstance = generatorType match {
+              case "injected" => injectedRoutesGeneratorModule.get(null)
+              case "static" => staticRoutesGeneratorModule.get(null)
+              case _ => throw new Exception(s"Unrecognized generator type: $generatorType. Use injected or static")
+            }
+            val result = compileMethod.invoke(null,
+              routesCompilerTaskInstance,
+              routesGeneratorInstance,
+              generatedDir)
+            // compile method returns an object of type Either[Seq[RoutesCompilationError], Seq[File]]
+            result.getClass.getName match {
+              case "scala.util.Right" =>
+                val files = cl.loadClass("scala.util.Right")
+                  .getMethod("value")
+                  .invoke(result)
+                val asJavaMethod = cl.loadClass("scala.collection.convert.DecorateAsJava")
+                  .getMethod("seqAsJavaListConverter", cl.loadClass("scala.collection.Seq"))
+                val javaConverters = cl.loadClass("scala.collection.JavaConverters$")
+                val javaConvertersInstance = javaConverters.getField("MODULE$").get(javaConverters)
+                val filesJava = cl.loadClass("scala.collection.convert.Decorators$AsJava")
+                  .getMethod("asJava")
+                  .invoke(asJavaMethod.invoke(javaConvertersInstance, files))
+                  .asInstanceOf[java.util.List[File]]
+                Right(filesJava.asScala)
+              case "scala.util.Left" =>
+                // TODO: convert the error of type RoutesCompilationError to a CompilationResult
+                Left(Seq(CompilationResult(Path(""), PathRef(Path("")))))
```

Comment:
I'm not quite sure about the return type... Do we want to return a `CompilationResult` ?

Reply:
If it failed, we can either return a `Option`, `Either`, or `mill.eval.Result.Error`/`mill.eval.Result.Exception`. I don't care much either way, except returning an invalid `CompilationResult` is probably unacceptable since anyone trying to use it will find garbage inside